src/index-format/union_find.ml

```diff
@@ -0,0 +1,38 @@
+type 'a content =
+  | Root of { mutable value : 'a; mutable rank : int }
+  | Link of { mutable parent : 'a element }
+and 'a element = { mutable content : 'a content }
```

Comment:
Why not just defining an element as a `ref` (since it is exactly the same type)? 

`and 'a element = 'a content ref`

Maybe you will add, in the future, more information?>